gravpm.c

```diff
@@ -276,11 +276,16 @@ void powerspectrum_compute(const int64_t k2, const int kpos[3], pfft_complex * c
     /* Measure power spectrum: we don't want the zero mode.
      * Some modes with k_z = 0 or N/2 have weight 1, the rest have weight 2.
      * This is because of the symmetry of the real fft. */
-    int kint = floor(sqrt(k2));
-    if(kint >= 0 && kint < PowerSpectrum.size) {
+    if(k2 > 0) {
```

Comment:
The variance is large - but this doesn't matter if you are trying to measure relative changes due to cosmology. In circumstances where I care about the variance I actually rebin the power spectrum to variable length bins guaranteed to have enough modes. But I don't like to do that all the time as it has hidden an interesting bug at least once before.

Also I don't use modes with k< 4/BoxSize usually.